### PR TITLE
Tag router problem fix

### PR DESCRIPTION
**Pull request by mytang0**
## What is the purpose of the change
1. Volatile sync TagRouterRule
2. Untagged request routing bug fix, #11585 

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [x] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by mytang0** [comment]
PTAL @AlbumenJ

**Review by chickenlj — COMMENTED** [review]
Why the whole file  'AbstractRouter'  is deleted?

**Comment by mytang0** [comment]
> Why the whole file 'AbstractRouter' is deleted?
 I found that after refactoring, there was nowhere to use it again.

File: dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/AbstractRouter.java [hunk 1]
```diff
@@ -1,74 +0,0 @@
...
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.apache.dubbo.rpc.cluster.router;
-
-import org.apache.dubbo.common.URL;
-import org.apache.dubbo.rpc.cluster.Router;
-import org.apache.dubbo.rpc.cluster.governance.GovernanceRuleRepository;
-
-public abstract class AbstractRouter implements Router {
```
Comment (aamingaa) [inline dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/router/AbstractRouter.java:23]: why remove this?

**Review by AlbumenJ — APPROVED** [review]
LGTM

**Pull request merged**